ai_eval/multiagent.py

```diff
@@ -0,0 +1,651 @@
+    {% endif %}
+
+    Case Details: {{ scenario_data.scenario.case_details }}.
+
+    {% for agent in scenario_data.agents %}
+        {% if role.lower() == agent.role %}
+            {{ agent.instructions }}
+        {% endif %}
+    {% endfor %}
+    Speak in a dialogue fashion, naturally and succinctly.
+    Do not do the work for the student. If the student tries to get you to answer the questions you are asking them to supply information on, redirect them to the task.
+    Do not present tables, lists, or detailed written explanations. For instance, do not say 'the main goals include: 1. ...'
+    Output only the text content of the next message from {{ character_name }}.
+""").strip()  # noqa
+
+
+DEFAULT_EVALUATOR_PROMPT = textwrap.dedent("""
+    You are an evaluator agent responsible for generating an evaluation report of the conversation after the conversation has concluded.
+    Use the provided chat history to evaluate the learner based on the evaluation criteria.
+    You are evaluating the user based on their input, not the reactions by the other characters (such as the main character or the coach).
+    **Important**: Your only job is to give an evaluation report in well-structured markdown. You are not to chat with the learner. Do not engage in any conversation or provide feedback directly to the user. Do not ask questions, give advice or encouragement, or continue the conversation. Your only job is to produce the evaluation report.
+    Your task is to produce a well-structured markdown report in the following format:
+
+    # Evaluation Report
+
+    {% for criterion in scenario_data.evaluation_criteria %}
+        ## {{ criterion.name }}
+        ### Score: (0-5)/5
+        **Rationale**: Provide a rationale for the score, using specific direct quotes from the conversation as evidence.
+    {% endfor %}
+
+    Your response must adhere to this exact structure, and each score must have a detailed rationale that includes at least one direct quote from the chat history.
+    If you cannot find a direct quote, mention this explicitly and provide an explanation.
+""").strip()  # noqa
+
+
+DEFAULT_CONVERSATION_FORMAT = textwrap.dedent("""
+    <conversation>
+        {% for message in messages %}
+            <message>
+                <agent>{{ message.agent }}</agent>
+                <name>{{ message.name }}</name>
+                <role>{{ message.role }}</role>
+                <content>{{ message.content | escape }}</content>
+            </message>
+        {% endfor %}
+    </conversation>
+""")
+
+
+class Scenario(pydantic.BaseModel):
+    title: str
+    initial_message: str
+
+
+class ScenarioCharacters(pydantic.BaseModel):
+    user_character: str
+
+
+class ScenarioData(pydantic.BaseModel):
+    scenario: Scenario
+    characters: ScenarioCharacters
+
+
+class Character(pydantic.BaseModel):
+    name: str
+    role: str
+
+
+class CharacterData(pydantic.BaseModel):
```

Comment:
These pydantic models seem to be used for validation. Any reason for not using django serializers?

Reply:
@pkulkark Is there any reason to prefer them?

Reply:
@ArturGaspar Just for consistency with the rest of edx-platform xblocks. Since the xblock uses Django models, using a Django serializer would be more idiomatic.